common/djangoapps/student/views.py

```diff
@@ -1230,30 +1233,43 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
 
         # if we didn't find this username earlier, the account for this email
         # doesn't exist, and doesn't have a corresponding password
+        msg = None
         if username != "":
             if settings.FEATURES['SQUELCH_PII_IN_LOGS']:
                 loggable_id = user_found_by_email_lookup.id if user_found_by_email_lookup else "<unknown>"
                 AUDIT_LOG.warning(u"Login failed - password for user.id: {0} is invalid".format(loggable_id))
             else:
                 AUDIT_LOG.warning(u"Login failed - password for {0} is invalid".format(email))
 
-        # Labster changes
-        # If there was login error, suggest student to login to region server
-        regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
-        # List of country codes: https://dev.maxmind.com/geoip/legacy/codes/iso3166/
-        current_region = request.session.get('country_code')
-        if current_region in regions.keys():
-            region_info = regions[current_region]
-            msg = _(
-                'If you are a student in {region_name}, your university may now be using the {region_code} version of '
-                'Labster. Please try to log into <a href="{region_login_url}">{region_name}</a> region.'
-            ).format(
-                region_name=region_info['name'],
-                region_code=region_info['region_code'],
-                region_login_url=region_info['login_url'],
-            )
+        # Start: Added by Labster
         else:
-            msg = _('Email or password is incorrect.')
+            # If there was login error, check whether user with the given email exists in one of the regions. If yes,
+            # redirect to the appropriate login url or suggest student to login to the appropriate region server.
+            try:
+                region = get_region_user(request, email, enrolled=False)
+            except UserAlreadyExistError as exc:
+                msg = _(
+                    'It appears that your account is located on our {region_code} server. '
+                    'Please log in <a href="{login_url}">here</a> instead.'
+                ).format(
+                    region_code=exc.region['region_code'],
+                    login_url=exc.region['login_url'],
+                )
+                region = None
```

Comment:
i'd do something like this instead to make it more explicit:
```py

def is_new_user(...):
  ...

is_new_user.UserAlreadyExistError = UserAlreadyExistError

...

base_url = request.build_absolute_uri('/')[:-1]
regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
try:
   is_new_user(base_url, regions, email, enrolled=False)
except is_new_user.UserAlreadyExistError as exc:
  msg = ...
else:
  if settings.LABSTER_FEATURES.get('ENABLE_REGION_IPADDR_WARNING'):
    current_region = request.session.get('country_code')
    region = regions.get(current_region) 
    if region: 
       msg = ...
  msg = msg or _("Account doesn't exist.")
```
or even without exceptions
```py
base_url = request.build_absolute_uri('/')[:-1]
regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
region = get_user_region(base_url, regions, email, enrolled=False)
if region:
    msg = ...
elif settings.LABSTER_FEATURES.get('ENABLE_REGION_IPADDR_WARNING'):
    current_region = request.session.get('country_code')
    region = regions.get(current_region) 
    if region: 
       msg = ...
else:
       msg = _("Account doesn't exist.")
```

Reply:
I agree to using second option, I think it'll be more simple